datum/settings/datum_gateway_config.json

```diff
@@ -0,0 +1,15 @@
+{
+    "bitcoind":{
+        "rpcuser": "enter your bitcoin RPC user",
+        "rpcpassword": "enter your bitcoin RPC addres",
+        "rpcurl": "172.17.0.1:9332"
+    },
+    "api":{
+         "listen_port": 21000
+    },
+    "mining":{
+        "pool_address": "enter your bitcoin address",
+        "coinbase_tag_primary": "enter your block tag",
+        "coinbase_unique_id": 1
+    }
+}
```

Comment:
So since the UI is just a read-only dashboard with no ability to adjust configuration, it sounds like we'll be shipping the app with sensible defaults that will make sense for most users. And then we can provide instructions for how to update config via CLI (e.g., naming blocks).

Do you know if any of this is configurable by environment variables in the datum container?

We could take advantage of the existing Knots environment variables and do something really simple so users are configured to Knots on install:
```yaml
datum:
  image: ghcr.io/retropex/datum:v1.3@sha:<digest>
  environment:
    RPC_USER: "${APP_BITCOIN_KNOTS_RPC_USER}"
    RPC_PASSWORD: "${APP_BITCOIN_KNOTS_RPC_PASS}"
    BITCOIN_HOST: "${APP_BITCOIN_KNOTS_NODE_IP}"
    RPC_PORT: "${APP_BITCOIN_KNOTS_INTERNAL_RPC_PORT}"
```

Ideally, we'd pass in these env vars which would tell Datum how to create the initial config file without us having to supply one from the host on install.

Reply:
No worries @Retropex, leave that one to me! I'm thinking what we'll do is create a `pre-install` script that writes out the `dataum_gateway_config.json` with the users bitcoind config, so that when they install the app they are fully connected to Knots without having to mess around.

And then because it is a `pre-install`-only script, it won't overwrite this file on app restart or update

Reply:
Are you okay if I push some commits?

Reply:
sure!

Reply:
Added a pre-start script here that configures to Knots RPC: https://github.com/getumbrel/umbrel-apps/pull/1661/commits/92b6554ce7dda21608ecd2440c1515ea4ab1b864